test/preview/components/generators/test_hf_utils.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from haystack.preview.components.generators.hf_utils import check_generation_params
+
+
+@pytest.mark.unit
+def test_empty_dictionary():
+    # no exception raised
+    check_generation_params({})
+
+
+@pytest.mark.unit
+def test_valid_generation_parameters():
+    # these are valid parameters
+    kwargs = {"max_new_tokens": 100, "temperature": 0.8}
+    additional_accepted_params = None
+    check_generation_params(kwargs, additional_accepted_params)
+
+
+@pytest.mark.unit
+def test_invalid_generation_parameters():
+    # these are invalid parameters
+    kwargs = {"invalid_param": "value"}
+    additional_accepted_params = None
+    with pytest.raises(ValueError):
+        check_generation_params(kwargs, additional_accepted_params)
+
+
+@pytest.mark.unit
+def test_additional_accepted_params_empty_list():
+    kwargs = {"temperature": 0.8}
+    additional_accepted_params = []
+    check_generation_params(kwargs, additional_accepted_params)
+
+
+@pytest.mark.unit
+def test_additional_accepted_params_known_parameter():
+    # both are valid parameters
+    kwargs = {"temperature": 0.8}
+    additional_accepted_params = ["max_new_tokens"]
+    check_generation_params(kwargs, additional_accepted_params)
+
+
+@pytest.mark.unit
+def test_additional_accepted_params_unknown_parameter():
+    kwargs = {"temperature": 0.8}
+    additional_accepted_params = ["valid_param"]
+    # Although valid_param is not generation param the check_generation_params
+    # does not raise exception because valid_param is passed as additional_accepted_params
+    check_generation_params(kwargs, additional_accepted_params)
```

Comment:
What I'm not understanding here is:
why do we check `temperature`?
:thinking: 

(maybe I misunderstood)

Reply:
Aha, yeah checking one valid parameter- temperature and one invalid in the test below. Just to show that it doesn't make a difference. Maybe I'm just going overboard here 😅

Reply:
I would change this test to something like:

```python
@pytest.mark.unit
def test_additional_accepted_params_unknown_parameter():
    kwargs = {"strange_param": "value"}
    additional_accepted_params = ["strange_param"]
    # Although strange_param is not generation param the check_generation_params
    # does not raise exception because strange_param is passed as additional_accepted_params
    check_generation_params(kwargs, additional_accepted_params)
```

Otherwise, this test looks almost identical to  `test_additional_accepted_params_known_parameter`.
(The fact that `additional_accepted_params` contains a known parameter or an unknown parameter makes no difference, because this variable is never checked.)